### PR TITLE
Added indicator extraction from word documents and improved all indicator extractions to utilize extractIndicators

### PR DESCRIPTION
**Pull request by idovandijk**
## Description

•	Updated indicator extraction in ExtractIndicatorsFromTextFile script to use extractIndicators command.
•	Updated indicator extraction in ReadPDFFile script to use extractIndicators command.
•	Cleaned up and extended Extract Indicators From File playbook, to utilize the new ability to extract indicators from word documents.
•	Fixed a bug where ".eml" and ".msg" files were skipped in one task, but not the task following that task.
•	Added a whole new script that uses a dedicated and customized docker image, to convert doc to docx, and also parse those files for indicators. The script extracts indicators both from paragraphs, from tables AND from core properties of the file.
•	Fixed bugs in update_id_set.py (with Rony)
•	Greatly improved `Extract Indicators From File - Generic - test` playbook. It now tests DOC, DOCX, TXT and PDF files. It thoroughly tests indicator extraction by looking for certain indicators that are found within the files. The default test for ReadPDFFile was replaced with this one.

## Related Issues
fixes: demisto/etc#14542
fixes: demisto/etc#13588
fixes: https://github.com/demisto/etc/issues/15025

### CODE REVIEW

**Comment by idovandijk** [comment]
@JonathanMeler can you approve this? We also have some fixes for Rony's scripts here

**Comment by idovandijk** [comment]
@jonathanm green build as far as my tests go. We can do a CR once I go over with alex

File: Scripts/script-ExtractIndicatorsFromWordFile.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+          self.all_data = ""
+
+      def get_file_details(self):
+          cmd_res = demisto.executeCommand('getFilePath', {'id': demisto.args()['entryID']})
+          file_res = cmd_res[0]
+          if isError(file_res):
+              file_res["Contents"] = "Error fetching entryID: " + file_res["Contents"]
+              self.res = file_res
+          else:  # Got file path:
+              self.file_path = file_res['Contents']['path']
```
Comment (JonathanMeler) [inline Scripts/script-ExtractIndicatorsFromWordFile.yml]: Use get

File: Scripts/script-ExtractIndicatorsFromWordFile.yml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+              self.file_name = file_res['Contents']['name']
+
+      def convert_doc_to_docx(self):
```
Comment (JonathanMeler) [inline Scripts/script-ExtractIndicatorsFromWordFile.yml]: Add comment of dependencies

File: Scripts/script-ExtractIndicatorsFromWordFile.yml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+type: python
+tags:
+- parser
+- autoextract
+- doc
+- docx
+- word
+comment: |-
+  Used to extract indicators from Word files (DOC, DOCX).
+  The script does not extract data from macros (e.g., embedded code).
```
Comment (JonathanMeler) [inline Scripts/script-ExtractIndicatorsFromWordFile.yml]: Add reference on parsed data

File: Scripts/script-ExtractIndicatorsFromWordFile.yml [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+              self.res = fileResult(output_file_name, f_data)
+
+      def extract_indicators(self):
+          try:
+              document = Document(self.file_path)
+              self.all_data = self.get_paragraphs(document)
+              self.all_data += self.get_tables(document)
+              self.all_data += self.get_core_properties(document)
+
+              output_file_name = self.file_name[0:self.file_name.rfind('.')] + '.txt'
```
Comment (JonathanMeler) [inline Scripts/script-ExtractIndicatorsFromWordFile.yml]: don't export to txt file. just print in war-room

File: Scripts/script-ExtractIndicatorsFromWordFile.yml [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+              self.extract_indicators()
+          elif self.file_name.endswith(".docx"):
+              self.extract_indicators()
+          else:
+              return_error("Input file is not a doc file.")
+
+  parser = WordParser()
+  parser.parse_word()
+  # Extract indicators (omitting context output, letting auto-extract work)
+  demisto.results(str(demisto.executeCommand("extractIndicators", {
```
Comment (JonathanMeler) [inline Scripts/script-ExtractIndicatorsFromWordFile.yml]: Return as human readable

Comment (JonathanMeler) [reply Scripts/script-ExtractIndicatorsFromWordFile.yml]: Return all data
Indicators
errors
Parsed data

File: Scripts/script-ReadPDFFile.yml [hunk 6]
```diff
@@ -94,10 +84,25 @@ script: |-
...
+              for k,v in metadata.iteritems():
+                  all_pdf_data += str(v)
+          if text:
+              all_pdf_data += text
+          if URLs:
+              for u in URLs:
+                  u = u["Data"] + " "
+                  all_pdf_data += u
+              # Extract indicators (omitting context output, letting auto-extract work)
+          demisto.results(str(demisto.executeCommand("extractIndicators", {
```
Comment (JonathanMeler) [inline Scripts/script-ReadPDFFile.yml]: Return Human readable

File: Scripts/script-ExtractIndicatorsFromWordFile.yml (hunk 5 above)
Comment (idovandijk) [reply Scripts/script-ExtractIndicatorsFromWordFile.yml]: OK, but why both the indicators and the full data?

**Comment by idovandijk** [comment]
@ronykoz Please, when you have time, take a look at `id_set.json`, I resolved some conflicts myself and need another pair of eyes to see that I didn't ruin anything there

File: Tests/id_set.json [hunk 7]
```diff
@@ -4761,6 +4758,76 @@
...
+                    "ExtractURL", 
+                    "getFilePath"
+                ], 
+                "tests": [
+                    "Extract Indicators From File - test"
+                ]
+            }
+        }, 
+        {
+            "ExtractIndicatorsFromWordFile": {
```
Comment (ronykoz) [inline Tests/id_set.json]: remove this one, its a dup

Comment (idovandijk) [reply Tests/id_set.json]: Thanks! Done

**Comment by idovandijk** [comment]
@anara123 Need merge please. Failing tests are not mine.

**Pull request merged**